cli/lock.ml

```diff
@@ -208,15 +212,24 @@ let get_pin_depends ~global_state local_opam_files =
   let open Rresult.R.Infix in
   root_pin_depends local_opam_files >>= pull_pin_depends ~global_state
 
+let interpret_solver_error ~switch_state = function
+  | `Msg _ as err -> err
+  | `Diagnostics d ->
+      (match Opam_solve.not_buildable_with_dune d with
+      | [] -> ()
+      | offending_packages ->
+          check_dune_universe_repo ~switch_state offending_packages);
```

Comment:
I agree, though that would require quite a bit of restructuring though if we don't just do `Fmt.str "%s\n%s" diagnostics our_warning` which personally I find rather ugly. Then this function would need to return instead of an ``[> `Msg of string]`` something more elaborate that the caller can then display the diagnostic and our warning separately.